File: Java/BubleSort.java
package Java;

import java.util.Arrays;

public class BubleSort {
    public static void main(String[] args) {
        int[] array = { 1, 4, 35, 3, 2, 5 };
        Bubble(array);
        System.out.println(Arrays.toString(array));

    }

    public static void Bubble(int[] array) {
        boolean swapped;
        // It's take the array and compare to it's previous number and swap if required
        for (int i = 0; i < array.length; i++) {
            // If element is already in order don't touch it
            swapped = false;
            for (int j = 1; j < array.length - i; j++) {
                System.out.println(Arrays.toString(array));
                System.out.println((array[j]));
                System.out.println((array[j - 1]));
                if (array[j - 1] > array[j]) {
                    int temp = array[j - 1];
                    array[j - 1] = array[j];
                    array[j] = temp;
                    swapped = true;
                }
            }
            if (!swapped) {
                break;
            }
        }
    }
}

File: Js/BubleSort.js
let bubble = (arr) => {
  for (let i = 0; i < arr.length; i++) {
    // Here the thing is to  itration  over the array
    let swapped = false
    for (let j = 1; j < arr.length - i; j++) {
      console.log(arr)
      if (arr[j - 1] > arr[j]) {
        swapped = true
        let temp = arr[j - 1]
        arr[j - 1] = arr[j]
        arr[j] = temp
      }
    }
    if (!swapped) {
      break
    }
  }
}

let arr = [32, 1, 4, 3, 5, 0]
bubble(arr)
console.log(arr)
